com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/RenameOldAppsStep.java

```diff
@@ -0,0 +1,101 @@
+import org.flowable.engine.delegate.DelegateExecution;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+
+import javax.inject.Named;
+import java.util.List;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.NavigableSet;
+import java.util.TreeSet;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+@Named("renameOldAppsStep")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class RenameOldAppsStep extends RenameAppsStep {
+
+    @Override
+    protected StepPhase executeStepInternal(ExecutionWrapper execution) {
+        DeployedMta deployedMta = StepsUtil.getDeployedMta(execution.getContext());
+        if (deployedMta == null) {
+            return StepPhase.DONE;
+        }
+
+        getStepLogger().debug("Renaming old applications");
+
+        CloudControllerClient client = execution.getControllerClient();
+
+        List<String> deployedAppsNames = deployedMta.getModules()
+                                                    .stream()
+                                                    .map(DeployedMtaModule::getAppName)
+                                                    .map(this::removeAppNameSuffix)
+                                                    .collect(Collectors.toList());
+
+        List<String> deployedAppsInMta = StepsUtil.getDeployedApps(execution.getContext())
+                                                  .stream()
+                                                  .map(CloudApplication::getName)
+                                                  .filter(appName -> deployedAppsNames.contains(removeAppNameSuffix(appName)))
+                                                  .collect(Collectors.toList());
+
+        List<String> appsToProcess = getAppsToProcess(deployedAppsInMta, client);
+
+        for (String appName : appsToProcess) {
+            getStepLogger().debug("Renaming application " + appName);
+            client.updateApplicationName(appName, appName + OLD_SUFFIX);
+        }
+        return StepPhase.DONE;
+    }
+
+    private List<String> getAppsToProcess(List<String> deployedAppsInMta, CloudControllerClient client) {
+        NavigableSet<String> appsToProcess = new TreeSet<>(deployedAppsInMta);
+        Set<String> appsToDelete = new HashSet<>();
+
+        for (String appName : appsToProcess) {
+            String nextAppName = appsToProcess.higher(appName);
+            if (nextAppName != null && removeAppNameSuffix(appName).equals(removeAppNameSuffix(nextAppName))) {
+                String appToDelete = getAppToDelete(appName, nextAppName);
+                getStepLogger().debug("Found two versions of app " + appToDelete + ". Deleting older one");
+                appsToDelete.add(appToDelete);
+            }
+        }
+        appsToProcess.removeAll(appsToDelete);
+        CompletableFuture.runAsync(() -> deleteApplications(appsToDelete, client));
```

Comment:
`getAppsToProcess` should not delete applications. As its name suggests, it should only get the applications that must be processed.